vere/http.c

```diff
@@ -2554,45 +2579,52 @@ _proxy_parse_sni(const uv_buf_t* buf_u, c3_c** hot_c)
   return u3_pars_good;
 }
 
-/* _proxy_parse_ship(): determine destination for proxied request
+/* _proxy_parse_ship(): determine destination (unit ship) for proxied request
 */
 static u3_noun
 _proxy_parse_ship(c3_c* hot_c)
 {
-  u3_noun sip = u3_nul;
-  c3_c* dom_c;
-
   if ( 0 == hot_c ) {
-    return sip;
-  }
-
-  dom_c = strchr(hot_c, '.');
-
-  if ( 0 == dom_c ) {
-    return sip;
+    return u3_nul;
   }
+  else {
+    c3_c* dom_c = strchr(hot_c, '.');
 
-  // XX revisit
-  c3_assert( 0 != u3_Host.sam_u.dns_c );
-
-  c3_w dif_w = dom_c - hot_c;
-  c3_w dns_w = strlen(u3_Host.sam_u.dns_c);
-
-  if ( (dns_w != strlen(hot_c) - (dif_w + 1)) ||
-       (0 != strncmp(dom_c + 1, u3_Host.sam_u.dns_c, dns_w)) ) {
-    return sip;
-  }
+    if ( 0 == dom_c ) {
+      return u3_nul;
+    }
+    else {
+      //  length of the first subdomain
+      //
+      c3_w dif_w = dom_c - hot_c;
+      c3_w dns_w = strlen(PROXY_DOMAIN);
+
+      //  validate that everything after the first subdomain
+      //  matches the proxy domain
+      //  (skipped if networking is disabled)
+      //
+      if ( (c3y == u3_Host.ops_u.net) &&
```

Comment:
So if we're in local-only mode, this means we skip to the else... Do we run the proxy when we're in local-only mode?

Reply:
Not by default, but we can (if it's set in the http-config %eyre effect %form).